adapters/onemobile/onemobile.go

```diff
@@ -0,0 +1,189 @@
+package onemobile
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type OneMobileAdapter struct {
+	http *adapters.HTTPAdapter
+	URI  string
+}
+
+func (a *OneMobileAdapter) Name() string {
+	return "onemobile"
+}
+
+func (a *OneMobileAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func (a *OneMobileAdapter) MakeRequests(requestIn *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	request := *requestIn
```

Comment:
The `requestIn` you are given is already a shallow copy of the original request, unique to your adapter. I don't think you need to make another copy. Along with that, the `Imp`s are also all shallow copies unique to your adapter. You still do need to copy the `Site` object as you do below.

Also seems a bit unclear and arbitrary where you use `requestIn` and `request` below, so dropping the copy might help clear that up.

Reply:
OK, if is a copy already I'll clean it.